canary/generator/download/download_whole.py

```diff
@@ -0,0 +1,38 @@
+from collections import defaultdict
+import json
+from json import JSONDecodeError
+from ast import literal_eval
+import sys
+import os
+
+import requests
+
+from .query import query
+
+if __name__ == '__main__':
+
+    dates_versions = requests.get(
+        'https://aggregates.telemetry.mozilla.org/aggregates_by/build_id/channels/nightly/dates/').text
+    dates_versions = literal_eval(dates_versions)
+    dates_for_each = defaultdict(list)
+    for item in dates_versions:
+        dates_for_each[item['version']].append(item['date'])
+
+    for v in dates_for_each.keys()[-5:]:
```

Comment:
Can we use the more readable `version` here?
Where does the `-5` come from? A comment on that should help here.

Reply:
Maybe I'll add an argument?